service/journal/service.go

```diff
@@ -1,96 +1,83 @@
 package journal
 
 import (
-	"fmt"
 	"github.com/labstack/echo/v4"
 	"github.com/matteoaricci/jot-api/models/journal"
+	"github.com/matteoaricci/jot-api/repo"
 	"net/http"
 	"strconv"
 )
 
-var existingJournals = []models.JournalVM{{
-	Title:       "Psychopomp",
-	Description: "Japanese Breakfast's first album",
-	ID:          "1",
-}, {
-	Title:       "Soft Sounds from Another Planet",
-	Description: "Absolute banger followup",
-	ID:          "2",
-}, {
-	Title:       "Jubilee",
-	Description: "Here Michelle Zauner asks: what if joy was as complex as grief",
-	ID:          "3",
-}}
-
-func Delete(id string) ([]models.JournalVM, *echo.HTTPError) {
-	j := findJournal(id)
-	if j == nil {
-		return nil, &echo.HTTPError{
-			Code:    http.StatusNotFound,
-			Message: fmt.Sprintf("Unable to find journal with id %s", id),
+func Delete(id string) *echo.HTTPError {
+	err := repo.DeleteJournal(id)
+	if err != nil {
+		if err.Error() == "record not found" {
```

Comment:
suggestion: instead of relying on this text never changing, it's better practice to use the `errors.Is` function to check an error. This one in particular originates from the gorm logger package

```suggestion
		if errors.Is(err, logger.ErrRecordNotFound) {
```

Not only does this protect you from if gorm decided to change the text, but it also detects the error even if it is wrapped up in another error.